tooling/generate-index.js

```diff
@@ -0,0 +1,39 @@
+
+const fs = require('fs');
+const path = require('path');
+
+// set directories and paths
+const methodsDir = path.join(__dirname, '../methods');
+const indexFile = path.join(methodsDir, 'index.json');
+
+// Process all method files and generate an index file
+(async () => {
+
+  const allMethods = [];
+  fs.readdirSync(methodsDir).forEach(filename => {
+    // skip index file
+    if(filename === 'index.json') {
+      return;
+    }
+
+    const methodFile = path.join(methodsDir, filename);
+    const methodData = fs.readFileSync(methodFile, 'utf-8');
+    console.log("Processing:", filename, methodData);
+
+    try {
+      const methodJson = JSON.parse(methodData);
+      allMethods.push(methodJson);
+    } catch(e) {
+        console.error('❌ ERROR: Failed to parse ', filename, e);
+        process.exit(1);
```

Comment:
@msporny these exits force the CI to fail, if building the index fails...this ensures PR previews will only function when CI is passing, which prevents us from needing to review anything that has not passed automatic checks.

Reply:
This is what a passing run looks like: https://github.com/w3c/did-spec-registries/runs/4131761620?check_suite_focus=true#step:6:8

Reply:
Sounds good... I forget the mechanism that publishes interim builds (the index.json file in them), but I know it exists, so happy to leave commits to index.json out of the repo.